### PR TITLE
[PM-6658] Migrate disableFavicon to Domain Settings service and remove Settings service

### PR DESCRIPTION
**Pull request by jprusik**
## Type of change

- Tech debt (refactoring, code cleanup, dependency upgrades, etc)

## Objective

migrate global `disableFavicon` setting to Domain settings state

## Code changes

- Create migration script
- replace usages of `disableFavicon` with `showFavicons` (with inverted logic where appropriate)
- Remove the Settings service, which is now no longer needed after these changes

### CODE REVIEW

File: apps/desktop/src/app/accounts/settings.component.ts [hunk 1]
```diff
@@ -115,9 +114,8 @@ export class SettingsComponent implements OnInit {
     private autofillSettingsService: AutofillSettingsServiceAbstraction,
     private messagingService: MessagingService,
     private cryptoService: CryptoService,
-    private modalService: ModalService,
```
Comment (jprusik) [inline apps/desktop/src/app/accounts/settings.component.ts:118]: unused

File: apps/web/src/app/settings/preferences.component.ts [hunk 2]
```diff
@@ -52,9 +51,8 @@ export class PreferencesComponent implements OnInit {
     private i18nService: I18nService,
     private vaultTimeoutSettingsService: VaultTimeoutSettingsService,
     private platformUtilsService: PlatformUtilsService,
-    private messagingService: MessagingService,
```
Comment (jprusik) [inline apps/web/src/app/settings/preferences.component.ts:55]: removed for not being in use

File: libs/angular/src/vault/components/icon.component.ts [hunk 3]
```diff
@@ -35,15 +35,15 @@ export class IconComponent implements OnInit {
...
 
   async ngOnInit() {
     const iconsUrl = this.environmentService.getIconsUrl();
 
     this.data$ = combineLatest([
-      this.settingsService.disableFavicon$.pipe(distinctUntilChanged()),
+      this.domainSettingsService.showFavicons$.pipe(distinctUntilChanged()),
       this.cipher$.pipe(filter((c) => c !== undefined)),
-    ]).pipe(map(([disableFavicon, cipher]) => buildCipherIcon(iconsUrl, cipher, disableFavicon)));
+    ]).pipe(map(([showFavicon, cipher]) => buildCipherIcon(iconsUrl, cipher, showFavicon)));
```
Comment (jprusik) [inline libs/angular/src/vault/components/icon.component.ts:47]: The logic inversion for this is handled within `buildCipherIcon`

**Review by LRNcardozoWDF — DISMISSED** [review]
LGTM!

File: libs/common/src/vault/icon/build-cipher-icon.ts [hunk 4]
```diff
@@ -2,12 +2,7 @@ import { Utils } from "../../platform/misc/utils";
 import { CipherType } from "../enums/cipher-type";
 import { CipherView } from "../models/view/cipher.view";
 
-export function buildCipherIcon(
-  iconsServerUrl: string,
-  cipher: CipherView,
-  isFaviconDisabled: boolean,
-) {
-  const imageEnabled = !isFaviconDisabled;
```
Comment (cagonzalezcs) [inline libs/common/src/vault/icon/build-cipher-icon.ts:10]: 👏 👏 👏

**Review by cagonzalezcs — APPROVED** [review]
Looks solid from my side, approving.

**Review by MGibson1 — DISMISSED** [review]

**Review by cagonzalezcs — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Review by LRNcardozoWDF — APPROVED** [review]

**Pull request merged**